### PR TITLE
[MHLO] Add torch-to-mhlo e2e support for aten.gather op

### PR DESCRIPTION
**Pull request by Vremold**

### CODE REVIEW

File: e2e_testing/xfail_sets.py [hunk 1]
```diff
@@ -24,6 +24,8 @@
 
 MHLO_PASS_SET = {
     "BroadcastToIdentityCaseStaticModule_basic",
+    "GatherStaticModule_basic",
+    "GatherModule_basic",
```
Comment (tanyokwok) [inline e2e_testing/xfail_sets.py:28]: Could you please add more dynamic shapes e2e tests to cover the following cases:
1. Both input and index are randomly generated; (The index could be un-ordered)
2. Test data with different ranks

**Review by tanyokwok — COMMENTED** [review]
Thanks @Vremold ! It LGTM with a minor comment on the e2e test.

File: e2e_testing/xfail_sets.py (hunk 1 above)
Comment (Vremold) [reply e2e_testing/xfail_sets.py:28]: Done.

**Review by tanyokwok — APPROVED** [review]
LGTM

**Comment by Vremold** [comment]
Merged. Thanks @tanyokwok @silvasean !

**Pull request merged**